source/common/upstream/upstream_impl.cc

```diff
@@ -102,6 +102,8 @@ parseClusterSocketOptions(const envoy::config::cluster::v3::Cluster& config,
                           const envoy::config::core::v3::BindConfig bind_config) {
   Network::ConnectionSocket::OptionsSharedPtr cluster_options =
       std::make_shared<Network::ConnectionSocket::Options>();
+  Network::Socket::appendOptions(cluster_options,
+                                 Network::SocketOptionFactory::buildSocketSigpipeOptions());
```

Comment:
If we have this here and on the listener side, do we still need the process-wide `signal()` call that attempts to ignore SIGPIPE? I suppose that would be a riskier change, but can you add a TODO above that call to consider removing it and maybe also add some comments here and on the listener side of why we set this?

Reply:
I think we'll always need the process-wide `signal()` call since this additional socket option only applies on certain OSes, as gated by `#ifdef SO_NOSIGPIPE` in this PR, so it probably doesn't make sense to have a todo. I updated with additional documentation as requested.